src/main/LibsodiumEncryptor.ts

```diff
@@ -36,14 +36,19 @@ export class LibsodiumEncryptor implements Encryptor {
     const nonce = randombytes_buf(crypto_secretbox_NONCEBYTES)
     const encryptionKey = this.deriveKey(dataKey, KeyType.ENCRYPTION)
 
-    const encryptedFields: { [P in K]: Uint8Array } = {} as any
+    const encryptedFields: {
+      [P in K]: Uint8Array
+    } = {} as ItemWithEncryptedFields<T, K>
 
     fieldsToEncrypt.forEach((fieldName) => {
-      encryptedFields[fieldName] = crypto_secretbox_easy(
-        this.toBuffer(item[fieldName]),
-        nonce,
-        encryptionKey
-      )
+      const fieldValue = item[fieldName]
+      if (fieldValue !== undefined && fieldValue !== null) {
```

Comment:
@teeeteee  -- I think it's slightly nicer if we handle the null/undefined case at the time of encryption rather than decryption. That way we don't end up encrypting null values